File: automobile/automobile-rental-angular/src/app/modules/customer/service/customer.service.ts
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { StorageService } from '../../../auth/service/storage/storage.service';

const BASIC_URL = ["http://localhost:9090"];

@Injectable({
  providedIn: 'root'
})
export class CustomerService {

  constructor(private http:HttpClient) { }

  getAllCars():Observable<any>{
    return this.http.get(BASIC_URL + "/api/customer/cars", {
      headers: this.createAuthorizationHeader() // Pass headers object

    })
  }

  getCarById(carId: number):Observable<any>{
    return this.http.get(BASIC_URL + "/api/customer/car/" + carId, {
      headers: this.createAuthorizationHeader() // Pass headers object

    })
  }

  bookACar(bookACarDto: any):Observable<any>{
    return this.http.post(BASIC_URL + "/api/customer/car/book",  bookACarDto, { 
      headers: this.createAuthorizationHeader()

    })
  }

  getBookingsByUserId(): Observable<any> {
    return this.http.get(BASIC_URL + "/api/customer/car/bookings/" + StorageService.getUserId(), {
      headers: this.createAuthorizationHeader()
    })
  }

  searchCar(searchCarDto: any): Observable<any>{
    return this.http.post(BASIC_URL + "/api/customer/car/search", searchCarDto, {
      headers: this.createAuthorizationHeader()
    });
  }


  createAuthorizationHeader(): HttpHeaders {
    let authHeaders: HttpHeaders = new HttpHeaders();
    return authHeaders.set(
      'Authorization',
      'Bearer ' + StorageService.getToken()
    );
  }
}

File: automobile/automobile-rental/src/main/java/com/CarPerHour/services/customer/CustomerService.java
package com.CarPerHour.services.customer;

import java.util.List;

import org.springframework.stereotype.Service;

import com.CarPerHour.dto.BookACarDto;
import com.CarPerHour.dto.CarDto;
import com.CarPerHour.dto.CarDtoListDto;
import com.CarPerHour.dto.SearchCarDto;

@Service
public interface CustomerService {
	
	List<CarDto> getAllCars();
	boolean bookACar(BookACarDto bookACarDto);
	CarDto getCarById(Long carId);
	List<BookACarDto> getBookingsByUserId(Long userId);
	CarDtoListDto searchCar(SearchCarDto searchCarDto);

}
